### ISSUE TITLE
Build breaks with OCaml 4.03

### ISSUE THREAD

**Issue by robblanco** [teyjus/teyjus#112]
The build process fails when using version 4.03 of the OCaml compiler. A reason for this is that several String functions have become deprecated, and the warnings are treated fatally by the build. You might want to ignore these warnings, given that the functions meant to replace them do not exist in previous versions, otherwise you will need to bump the compiler requirements all the way to the current version.

**Comment by markwright** [comment]
To build teyjus on Gentoo I have to rename 2 files, the following code fragment is from the src_prepare function in the ebuild here:

https://github.com/gentoo/gentoo/blob/master/dev-lang/teyjus/teyjus-2.1.ebuild

```
	if has_version ">=dev-lang/ocaml-4.03.0"; then
		# bug 591368
		pushd "${S}/source" || die
		sed -e 's@$(FNT)/ccode_stubs@$(FNT)/ccode_stubs_c@' \
			-e 's@\(FNT_ML_TO_C\[\] =\)@\1\n    $(FNT)/ccode_stubs@' \
			-i OMakefile || die
		cd "${S}/source/front" || die
		mv ccode_stubs.mli ccode_stubs.ml || die
		mv ccode_stubs.c ccode_stubs_c.c || die
		popd || die
	fi
```

to workaround this change in ocaml:

http://stackoverflow.com/questions/39698523/in-ocaml-4-03-0-ffi-fails-to-compile-with-no-implementations-provided-error/39701139#39701139

**Comment by diegoxfx** [comment]
2018 and the problem is still unsolved

**Comment by gisellemnr** [comment]
The first comment, by @robblanco, is fixed by the attached patch. Just remove the `.txt` extension.

[0001-Removing-deprecated-function-String.set.patch.txt](https://github.com/teyjus/teyjus/files/3435825/0001-Removing-deprecated-function-String.set.patch.txt)

**Comment by gisellemnr** [comment]
Fixed on 38772fa56ed7edef88e71df8a69eea4f341968ed
